lua/toggleterm/terminal.lua

```diff
@@ -98,15 +97,6 @@ local function setup_buffer_autocommands(term)
     },
   }
 
-  if conf.start_in_insert then
-    vim.cmd("startinsert!")
-    table.insert(commands, {
-      "BufEnter",
-      fmt("<buffer=%d>", term.bufnr),
-      "startinsert!",
```

Comment:
@JA-Bar is the issue resolved by removing the `!` from the command. It actually causes nvim to behave as if `A` was pressed, which isn't what I intended.

Reply:
Rather than adding this logic to the `opener`

Reply:
Oh you are correct! I didn't think about it being unintended, should be correct now.